src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -322,7 +331,36 @@ class AppsController(
     }
   }
 
-  private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = ???
+  private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = {
+    val maybeApp = groupManager.app(appId)
+    maybeApp.map { app =>
+      authorized(ViewRunSpec, app).apply {
+        val tasksF = instanceTracker.instancesBySpec flatMap { instancesBySpec =>
+          runningTasks(Set(appId), instancesBySpec)
+        }
+        onSuccess(tasksF) { tasks =>
+          complete(Json.obj("tasks" -> tasks.toRaml))
+        }
+      }
+    } getOrElse {
+      reject(Rejections.EntityNotFound.noApp(appId))
```

Comment:
Why not match the app and use `case None` here? I think that is easier to follow.